addons/payment/views/payment_templates.xml

```diff
@@ -373,8 +377,7 @@
 
     <!-- Verified token checkmark -->
     <template id="verified_token_checkmark" name="Payment Verified Token Checkmark">
-        <t t-if="0" name="payment_demo_hook"/>
-        <t t-else="">
+        <t t-if="token.provider_code!='demo'">
```

Comment:
There is a problem here: we usually avoid, as much as possible, to use code from other modules when they are not a dependency of the current module. Meaning that, here, you are checking for a `provider_code` that can't exist in `payment` if `payment_demo` isn't installed. That's why the previous code used a useless condition, which get overridden in `payment_demo`.

But truth be told, this check isn't needed anyway or needs to be done on the state of the provider instead, IMO.

The demo provider can't be in an "enable" state, meaning it'll always get the test token badge with this PR. I don't see why it'd be a problem to have the verified token check mark even with demo tokens, as it'll be with any other test tokens from other providers. But the task is quite unclear about this... I don't know if we should only show the test badge or test badge and verified token check mark... I mean, even in test state, it could be interesting to see if it's "verified" or not, or would it always be not verified for test tokens? :thinking: